resolwe_bio/processes/support_processors/rnaseqc_qc.py

```diff
@@ -0,0 +1,346 @@
+    IntegerField,
+    Persistence,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+STRAND_CODES = {
+    "IU": "non_specific",
+    "U": "non_specific",
+    "ISF": "forward",
+    "OSF": "forward",
+    "SF": "forward",
+    "ISR": "reverse",
+    "OSR": "reverse",
+    "SR": "reverse",
+}
+
+
+def format_UCSC(annotation_path):
+    """Format UCSC annotation to a format suitable for parsing with collapse_annotation.py script."""
+
+    def add_attributes(x):
+        new_attrs = ""
+        new_attrs += f'gene_name "{x.gene_id}"; '
+        new_attrs += f'gene_id "{x.gene_id}"; '
+        new_attrs += f'transcript_name "{x.gene_id}"; '
+        new_attrs += f'transcript_id "{x.gene_id}"; '
+        new_attrs += f'gene_type "{x.gene_id}"; '
+        new_attrs += f'transcript_type "{x.gene_id}"; '
+
+        new_attrs = new_attrs.strip()
+
+        return new_attrs
+
+    df = pd.read_csv(
+        annotation_path,
+        sep="\t",
+        comment="#",
+        header=None,
+        usecols=[0, 2, 3, 4, 6, 8],
+        low_memory=False,
+    )
+
+    df = pd.DataFrame(
+        {
+            "chromosome": df[0].str.lstrip("chr"),
+            "feature_type": df[2],
+            "start": df[3],
+            "end": df[4],
+            "strand": df[6],
+            "gene_id": df[8].str.extract(r'gene_id "(.*?)"', expand=False).values,
+        },
+    )
+
+    df.replace("", np.nan, inplace=True)
+
+    columns = {
+        "chromosome": "first",
+        "feature_type": "first",
+        "start": "min",
+        "end": "max",
+        "strand": "first",
+        "gene_id": "first",
+    }
+
+    df["gene_id"] = (
+        "chr"
+        + df["chromosome"].astype(str)
+        + "_"
+        + df["strand"].astype(str)
+        + "_"
+        + df["gene_id"].astype(str)
+    )
+
+    genes_df = df.groupby(["gene_id"]).agg(columns).assign(feature_type="gene")
+    transcripts_df = (
+        df.groupby(["gene_id"]).agg(columns).assign(feature_type="transcript")
+    )
+    df = pd.concat([genes_df, transcripts_df, df], ignore_index=True)
+
+    df["attributes"] = df.apply(lambda x: add_attributes(x), axis=1)
+
+    out_df = pd.DataFrame(
+        {
+            "chromosome": df["chromosome"],
+            "annotation_source": ["UCSC" for _ in range(len(df))],
+            "feature_type": df["feature_type"],
+            "start": df["start"],
+            "end": df["end"],
+            "score": ["." for _ in range(len(df))],
+            "strand": df["strand"],
+            "genomic_phase": ["." for _ in range(len(df))],
+            "attributes": df["attributes"],
+        },
+    )
+
+    out_df.sort_values(by=["chromosome", "start"], ascending=[True, True], inplace=True)
+    out_df.to_csv(
+        "formatted_annotation.gtf",
+        sep="\t",
+        header=False,
+        index=False,
+        quoting=csv.QUOTE_NONE,
+    )
+
+
+class QcRnaseqc(Process):
+    """RNA-SeQC QC analysis.
+
+    More information can be found in the
```

Comment:
Description could perhaps be more verbose. This is rephrased from the original publication as an example of a possible description:

> An efficient new version of RNA-SeQC that computes a comprehensive set of metrics for characterizing samples processed by a wide range of protocols. It also quantifies gene- and exon-level expression, enabling effective quality control of large-scale RNA-seq datasets.